webapp/src/components/call_widget/component.tsx

```diff
@@ -1502,6 +1508,12 @@ export default class CallWidget extends React.PureComponent<Props, State> {
                             },
                         },
                     });
+                    if (alertID === 'degradedCallQuality') {
+                        this.callQualityBannerLocked = true;
+                        setTimeout(() => {
+                            this.callQualityBannerLocked = false;
+                        }, DEGRADED_CALL_QUALITY_ALERT_WAIT);
```

Comment:
I'd expect this to be in the state to guarantee re-renders. I'm wondering if this is any different, though.

Reply:
The lock doesn't need to be in the state because setting it or unsetting it won't affect the UI, it's only the `setState` that it prevents/allows that will. Hope that makes it clearer?

Reply:
Ah, we keep the lock for the whole wait period. That makes sense, thank you!